src/proto/h1/role.rs

```diff
@@ -358,7 +358,17 @@ impl Http1Transaction for Server {
 
         let init_cap = 30 + msg.head.headers.len() * AVERAGE_HEADER_SIZE;
         dst.reserve(init_cap);
-        if msg.head.version == Version::HTTP_11 && msg.head.subject == StatusCode::OK {
+
+        #[cfg(feature = "http1_reason_phrase")]
+        let custom_reason_phrase = msg.head.extensions.get::<crate::ext::ReasonPhrase>();
+        #[cfg(not(feature = "http1_reason_phrase"))]
+        #[allow(non_upper_case_globals)]
+        const custom_reason_phrase: Option<()> = None;
```

Comment:
Here's where it starts getting a bit messy. To reduce the overall `cfg`ness of this code, I'm relying on `const` evaluation to ensure this does not have an effect on performance when the flag is not enabled.